tests/helpers/perf_run_helper.rb

```diff
@@ -944,75 +947,259 @@ def get_process_hash(perf_result_processes)
-      # compare results created in this run with latest baseline run
-      sql = "SELECT avg_cpu, avg_mem, avg_disk_write, avg_response_time, " \
-            "process_bolt_server_avg_cpu, process_bolt_server_avg_mem, " \
-            "process_ace_server_avg_cpu, process_ace_server_avg_mem, " \
-            "process_puppetdb_avg_cpu, process_puppetdb_avg_mem, " \
-            "process_console_services_release_avg_cpu, process_console_services_release_avg_mem, " \
-            "process_orchestration_services_release_avg_cpu, process_orchestration_services_release_avg_mem, " \
-            "process_puppet_server_release_avg_cpu, process_puppet_server_release_avg_mem " \
-            "FROM `perf-metrics.perf_metrics.atop_metrics` " \
-            "WHERE time_stamp = (" \
-          "SELECT MAX(time_stamp) " \
-        "FROM `perf-metrics.perf_metrics.atop_metrics` " \
-        "WHERE pe_build_number = '#{BASELINE_PE_VER}' AND test_scenario = '#{current_test_name}' " \
-           "GROUP BY pe_build_number, test_scenario)"
-
-      data = query_bigquery sql
-      if !data.empty?
-        logger.info("Baseline result returned from BigQuery: #{data}")
-      else
-        logger.error("Cannot find result that matches query: #{sql}")
-      end
-      data[0]
+      logger.error("Cannot find result that matches query: #{sql}")
+      result = {}
     end
+    return result
   end
 
   def baseline_assert(atop_result, gatling_result)
-    process_results = get_process_hash(atop_result.processes)
-
-    # Handle 3 different things: avg_response_time which comes from gatling result,
-    # global atop results and per process atop results
     baseline = get_baseline_result
-    baseline&.each do |key, value|
-      # orchestration has a jump in memory usage in 2019.2 due to plan executor
-      # https://tickets.puppetlabs.com/browse/BOLT-986
-      # Currently we don't have a way to whitelist that... so just turning off the check
-      # Ticket to make a way https://tickets.puppetlabs.com/browse/SLV-515
-      next if key.to_s == "process_orchestration_services_release_avg_mem" && BEAKER_PE_VER =~ /^2019.2*/
-
-      assert_value = if key.to_s == "avg_response_time"
-                       gatling_result.avg_response_time.to_f
-                     elsif key.to_s.start_with? "process"
-                       process_results[key.to_s].to_f
-                     else
-                       atop_result.send(key).to_f
-                     end
-
-      puts "Value for #{key} is baseline: #{value} Actual: #{assert_value}"
-      next if key.to_s.start_with?("process") && key.to_s.end_with?("cpu")
-
-      max_baseline_variance = \
-        if key.to_s == "process_orchestration_services_release_avg_mem"
-          15
-        else
-          10
-        end
-      next unless value.is_a? Integer
+    data = baseline_to_results_data(baseline, atop_result, gatling_result)
+    assert_later(validate_baseline_data(data), "Things are really fucked up now")
```

Comment:
I'm not offended by this language, but others may be, so we may want to keep it clean:
```suggestion
    assert_later(validate_baseline_data(data), "Things are really mucked up now")
```

Reply:
I have pushed a commit to change this message to something more appropriate to the context in which it gets emitted.